src/components/ha-selector/ha-selector-media.ts

```diff
@@ -0,0 +1,283 @@
+import { mdiPlayBox, mdiPlus } from "@mdi/js";
+import { css, CSSResultGroup, html, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import { fireEvent } from "../../common/dom/fire_event";
+import { supportsFeature } from "../../common/entity/supports-feature";
+import { getSignedPath } from "../../data/auth";
+import {
+  MediaClassBrowserSettings,
+  MediaPickedEvent,
+  SUPPORT_BROWSE_MEDIA,
+} from "../../data/media-player";
+import type { MediaSelector, MediaSelectorValue } from "../../data/selector";
+import type { HomeAssistant } from "../../types";
+import "../ha-alert";
+import "../ha-form/ha-form";
+import type { HaFormSchema } from "../ha-form/types";
+import { showMediaBrowserDialog } from "../media-player/show-media-browser-dialog";
+
+const MANUAL_SCHEMA = [
+  { name: "media_content_id", required: false, selector: { text: {} } },
+  { name: "media_content_type", required: false, selector: { text: {} } },
+];
+
+@customElement("ha-selector-media")
+export class HaMediaSelector extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public selector!: MediaSelector;
+
+  @property({ attribute: false }) public value?: MediaSelectorValue;
+
+  @property() public label?: string;
+
+  @property({ type: Boolean, reflect: true }) public disabled = false;
+
+  @state() private _manual = false;
+
+  @state() private _thumbnailUrl?: string | null;
+
+  willUpdate(changedProps) {
+    if (
+      changedProps.has("value") &&
+      this.value?.extra?.thumbnail !==
+        changedProps.get("value")?.extra?.thumbnail
+    ) {
+      if (
+        this.value?.extra?.thumbnail &&
+        this.value.extra.thumbnail.startsWith("/")
+      ) {
+        this._thumbnailUrl = undefined;
+        // Thumbnails served by local API require authentication
+        getSignedPath(this.hass, this.value.extra.thumbnail).then(
+          (signedPath) => {
+            this._thumbnailUrl = signedPath.path;
+          }
+        );
+      } else {
+        this._thumbnailUrl = this.value?.extra?.thumbnail;
+      }
+    }
+  }
+
+  protected render() {
+    const stateObj = this.value?.entity_id
+      ? this.hass.states[this.value.entity_id]
+      : undefined;
+
+    const supportsBrowse =
+      stateObj && supportsFeature(stateObj, SUPPORT_BROWSE_MEDIA);
+
+    return html`<ha-entity-picker
+        .hass=${this.hass}
+        .value=${this.value?.entity_id}
+        .label=${this.label ||
+        this.hass.localize("ui.components.selectors.media.pick_media_player")}
+        .disabled=${this.disabled}
+        include-domains='["media_player"]'
+        allow-custom-entity
+        @value-changed=${this._entityChanged}
+      ></ha-entity-picker>
+      ${this._manual || (stateObj && !supportsBrowse)
```

Comment:
Why check if stateObj is defined?

Reply:
If we dont have a state, we cant check if it supports browse, so assume it does

Reply:
You already guard for that when defining `supportsBrowse`

Reply:
yes, but if there is a state, and we checked if it supported browsing, and it didnt, then we get here.

If `supportsBrowse` is false because we didnt have a `stateObj`, we render the media browse UI

Reply:
Changed it be clearer